### ISSUE TITLE
An error occurs when calling nonapi.io.github.classgraph.concurrency.WorkQueue.runWorkLoop().

### ISSUE THREAD

**Issue by ahamana** [classgraph/classgraph#307]
After upgrading from 4.6.28 to 4.6.29, an error occurs when calling nonapi.io.github.classgraph.concurrency.WorkQueue.runWorkLoop().
\* The ClassGraph log is "classgraph-log.txt".

[log.zip](https://github.com/classgraph/classgraph/files/2823802/log.zip)

**Comment by lukehutch** [comment]
@ahamana がとうございます for the report. I just pushed what I think is a fix to git master. Can you please test this out?

**Comment by lukehutch** [comment]
I'm pretty sure this should be fixed now, so I pushed out the fix in 4.6.30. Thanks!

**Comment by ahamana** [comment]
@lukehutch I tested with 4.6.30, but an error occurs yet.
The stacktrace seem to be same.

[log-4.6.30.zip](https://github.com/classgraph/classgraph/files/2823983/log-4.6.30.zip)

**Comment by ahamana** [comment]
@lukehutch 
"newSingletonHolder.set(newInstance)" is done at finally block, so even if "newInstance" is null, newInstance is set to newSingletonHolder.

```
// Initialize newSingletonHolder with new instance of value.
V newInstance = null;
try {
    newInstance = newInstance(key, log);
    if (newInstance == null) {
        if (log != null) {
            log.log("newInstance returned null for key " + key);
        }
        throw new IllegalArgumentException("newInstance returned null for key " + key);
    }
} finally {
    // Have to call .set() even if an exception is thrown by newInstance(), or if newInstance
    // is null, since .set() calls initialized.countDown(). Otherwise threads that call .get()
    // can end up waiting forever.
    newSingletonHolder.set(newInstance);
}
```

So, there is a possibility that "singletonHolder.get()" returns null, I think.

```
if (singletonHolder != null) {
    // There is already a SingletonHolder in the map for this key -- get the value
    return singletonHolder.get();
}
```

**Comment by lukehutch** [comment]
Should be fixed in 4.6.31. Please comment again if not. Thank you!

**Comment by lukehutch** [comment]
Thanks for reporting and testing!
